### ISSUE TITLE
Feature request: create a panel buffer that lists all keys that were pressed while using Nyxt

### ISSUE THREAD

**Issue by pdelfino** [atlas-engineer/nyxt#2037]
This will be helpful for Nyxt Academy's videos. David Wilson from the YT channel _System Crafters_ uses a similar thing for his tutorials while teaching Emacs. If my understanding is correct, he uses [command-log-mode](https://github.com/lewang/command-log-mode).

@edgar-vincent also suggested using something like this so that people could see which keys I was pressing.

**Comment by aartaka** [comment]
`dispatch-input-event` can be a good starting point there :)

**Comment by Ambrevar** [comment]
@pdelfino Minor nit: no need to prefix your issue name with "Feature request" if we are going to use the `feature` tag.  That tag is more useful ;)

**Comment by pdelfino** [comment]
> @pdelfino Minor nit: no need to prefix your issue name with "Feature request" if we are going to use the `feature` tag. That tag is more useful ;)

Yes. We can add tags. But normal users **cannot**. Sometimes, other random users used this pattern (maybe influenced by me) to create titles like this. 

Examples: [1](https://github.com/atlas-engineer/nyxt/issues/1949), [2](https://github.com/atlas-engineer/nyxt/issues/1949), [3](https://github.com/atlas-engineer/nyxt/issues/1778), [4](https://github.com/atlas-engineer/nyxt/issues/1688), [5](https://github.com/atlas-engineer/nyxt/issues/1170)

This practice seems to have no downside. It might make the additions of labels easier (better for maintainers) and it might help with information retrieval - especially because we have multiple issues without any label. But I can stop if this bothers you... Maybe it looks ugly.

**Comment by Ambrevar** [comment]
It's OK for unprivileged users to prefix the title, I was just saying
this for you since you can add tags, it helps keeing titles shorter.

**Comment by pdelfino** [comment]
> It's OK for unprivileged users to prefix the title, I was just saying this for you since you can add tags, it helps keeing titles shorter.

I see and I was aware. I was just trying to help **promoting** the behavior among unprivileged users. But, I will keep the titles shorter from now on.

**Comment by Ambrevar** [comment]
Oh I see.

Then you gave me an idea: we can create a GitHub issue template for
feature requests.  By the way, maybe we could automatically tag them?

**Comment by Ambrevar** [comment]
Done in
https://github.com/atlas-engineer/nyxt/commit/6401f0686b4cfcd778d21ab9ee1507102fc0ae88
!

I guess now there really is no need for the "Feature request:" prefix.

**Comment by pdelfino** [comment]
> Oh I see. Then you gave me an idea: we can create a GitHub issue template for feature requests. By the way, maybe we could automatically tag them?

This is a great idea! Thanks :)

**Comment by hendursaga** [comment]
Not only the keys, but also the command, too, at least optionally!

**Comment by guojing0** [comment]
I think the prompt buffer also keeps some kinds of history of the commands used by an user?

**Comment by Ambrevar** [comment]
Indeed, there is the `history` prompt buffer command accessible with
`M-h` by default.

**Comment by aadcg** [comment]
Panel buffers were deleted in a4985675380150b931915cb79decf8b1029eaf60.

I don't think that this feature belong to Nyxt's core.
